CHANGELOG.md

```diff
@@ -3,7 +3,192 @@
+- Custom built-in function enhancement: let custom builtins halt evaluation ([#3534](https://github.com/open-policy-agent/opa/issues/3534))
+- Partial evaluation: Fix stack overflow on certain expressions ([#3559](https://github.com/open-policy-agent/opa/issues/3559))
+
+### Tooling
+
+- Query Profiling: `opa eval --profile` now supports a `--count=#` flag to gather metrics and profiling data over multiple runs, and displays aggregate statistics for the results ([#3651](https://github.com/open-policy-agent/opa/issues/3651)).
+
+  This allows you to gather more robust numbers to assess policy performance.
+
+- Docker images: Publish static image ([#3633](https://github.com/open-policy-agent/opa/issues/3633))
+
+  As of this release, you can use the staticly-built Linux binary from a docker image: `openpolicyagent/opa:v0.31.0-static`.
+  It contains the same binary that has been published since release v0.29.4, statically linked to musl, with evaluating Wasm disabled.
+
+### Fixes
+
+- Built-in `http.send`: ignore `tls_use_system_certs` setting on Windows. Having this set to _true_ (the default as of v0.29.0) would _always_ return an error on Windows.
+- Update query compiler to reject empty queries ([#3625](https://github.com/open-policy-agent/opa/issues/3625))
+- Partial Evaluation fix: Don't generate comprehension with unsafe variables ([#3557](https://github.com/open-policy-agent/opa/issues/3557))
+- Parser: modules containing _only_ tabs and spaces no longer lead to a runtime panic.
+- Wasm: ensure that the desired stack space for the C library calls (64KiB) is not reduced by data segments added in the compiler.
+   This is achieved by putting the stack first -- stack overflows now become "out of bounds" memory access traps.
+   Before, it would silently corrupt the static data.
+
+### Server and Runtime
+
+- New configuration for Management APIs: using `resource`, the request path for sending decision logs can be configured now ([#3618](https://github.com/open-policy-agent/opa/issues/3618)), authored by @[cbuto](https://github.com/cbuto)
+
+  `/logs` is still the default, but can now be overridden.
+  With this change, the `partition_name` config becomes deprecated, since its functionality is subsumed by this new configurable.
+
+### Documentation
+
+- How to debug? Clarify how to access `Note` events for debugging via explanations ([#3628](https://github.com/open-policy-agent/opa/issues/3628)) authored by @[enori](https://github.com/enori)
+- Clarify special characters for key, i.e. what `x["y"]` is necessary because `x.y` isn't valid ([#3638](https://github.com/open-policy-agent/opa/issues/3638)) authored by @[Hongbo-Miao](https://github.com/Hongbo-Miao)
+- Integrations: add Python Bottle library ([#3619](https://github.com/open-policy-agent/opa/pull/3619)), by @[dolevf](https://github.com/dolevf)
```

Comment:
While I don't mind this, I don't think we've listed added integrations in the past, as they aren't really attached to a release, and are immediately seen on the ecosystem page.

Reply:
Oops I wasn't aware that they turned up immediately. 💡 I'll remove this, thanks.